### ISSUE TITLE
Consider adding LRU cache to RocksDB impl of Database

### ISSUE THREAD

**Issue by nearmax** [near/nearcore#5203]
We haven't had an opportunity to optimize the performance of the RPC server and the ViewClient inside nearcore. Which leads to individual RPC nodes not being able to sustain the needed throughput, which in turn leads to high cloud expenses for running multiple nodes and the need to use throttling. 

Anecdotally, some people were able to optimize RPC server to achieve higher QPS by using Redis in nearcore: https://twitter.com/vgrichina/status/1458289037525934084 . We might be able to achieve QPS increase by simpler means as a temporary measure until we revisit RPC server and ViewClient implementations. I suggest we consider adding simple LRU cache from https://docs.rs/cached/0.26.2/cached/ directly into `get` and `write` methods of `impl Database for RocksDB` here:
https://github.com/near/nearcore/blob/734978677a7491743b5a2b4519b619f263e747ce/core/store/src/db.rs#L469-L563

`write` method would be responsible for invalidating LRU cache entries.

Unfortunately, AFAIR there is a suboptimality in `near-api-js` that forces it re-request all access from an account before signing a transaction. This will likely touch `iter_*` methods (likely `iter_prefix` method) of `impl Database for RocksDB`, which we might also want to wrap in cache. CC @MaximusHaximus , @frol 

We most likely would want to have this cache enabled only for the RPC nodes, and we should let the Contract Runtime team know about its existence, since it might affect the measurements of the param estimator.

**Comment by bowenwang1996** [comment]
Rocksdb has in-memory cache internally and we have it enabled.

**Comment by nearmax** [comment]
> Rocksdb has in-memory cache internally and we have it enabled.

Given the complexity of Rocksdb and that we revisited its configuration only once, I don't think we can be sure that we configured it correctly or that it is optimized for the view calls.

**Comment by bowenwang1996** [comment]
Indeed, we currently set the cache size for all columns to be 32mb https://github.com/near/nearcore/blob/6d8eee6ffbf6dedbae5022bd3d8acd2c2870a6b3/core/store/src/db.rs#L664, but not all columns are equal. I think we can and should allocate more for `ColState`. For example, we can do 256mb or even 512 for `ColState`. @mina86 @pmnoxx given that you worked with Rocksdb before, I wonder whether this is something you could help with.

**Comment by pmnoxx** [comment]
@bowenwang1996 Hmm, we could write some PR like this: https://github.com/near/nearcore/pull/5212

**Comment by ilblackdragon** [comment]
I would suggest to make this parameter a node configuration.

This way RPC nodes can configure a larger cache if they have more RAM compared to validator nodes or other nodes that don't need to constantly respond to view calls

**Comment by pmnoxx** [comment]
> node configuration


> I would suggest to make this parameter a node configuration.
> 
> This way RPC nodes can configure a larger cache if they have more RAM compared to validator nodes or other nodes that don't need to constantly respond to view calls

@ilblackdragon I can add it to `config.json`, is that what you meant?

**Comment by andrei-near** [comment]
Looks like this is already implemented https://github.com/near/nearcore/pull/6584

**Comment by wacban** [comment]
@nagisa & @darioush Is this what you're looking at now? If so can we close this?
